doc/research/05_acting.md

```diff
@@ -0,0 +1,199 @@
+
+## Lateral control
+
+Model: [Ackermann steering geometry](https://en.wikipedia.org/wiki/Ackermann_steering_geometry)
+
+### Pure Pursuit
+
+This is the simplest way to steer a vehicle, however this doesn't mean that it can't be used in practice. There are two main simplifications made.
+
+1. Use the bicycle model (combine front/rear wheels -> one wheel per axle)
+2. Elevation isn't considered (-> both axles are at the same height)
+
+The steering angle $\delta$ is defined as the angle of the front wheel to a line drawn between the center of both axles. This angle $\delta$ can also be defined as $tan(\delta) = L/R$ with $L$ as the wheelbase and $R$ the radius from the reference point (rear axle) to the Instantaneous Center of Rotation (ICR). Due to the bicycle model we can calculate $R = \frac{L}{tan(\delta)}$.
+
+![Bicycle Model with ICR](../assets/bicyclegeometry.png)
+*source: [2]*
+
+We now try to aim the circular arc to intersect with a point on our trajectory. This target point is always a defined distance (look ahead distance $l_d$) away from our reference point (dangling carrot). This leads to the following relation:
+
+![Dangling carrot geometry](../assets/danglingcarrotgeometry.png)
+*source: [2]*
+
+$\frac{l_d}{sin(\alpha)}= 2R$, where $\alpha$ is the current heading error. Combining the two equations leads to our desired steering angle.
+
+$$
+\delta(t) = arctan(2L * \frac{sin(\alpha)}{l_d})
+$$
+
+This method can be tuned to work perfect for a certain speed, however we would like it to work in all speed regimes. We therefore calculate $l_d$ based on our current speed $v(t)$ and a tunable constant $K_d$ (, $l_d(t) = v(t) * K_d$). This leads to the final equation for our steering angle.
+
+$$
+\delta(t) = arctan(2L* \frac{sin(\alpha)}{K_d * v})
+$$
+
+- simple controller
+- ignores dynamic forces
+- assumes no-slip condition
+- possible improvement: vary the look-ahead distance based on vehicle velocity
+- not really suited for straights
+
+### Stanley
+
+The Stanley controller, named after an autonomous offroad race car, takes the front axle as a reference, while still using the bicycle model. In addition to looking at the heading error $\psi$, close to what pure pursuit does, stanley also looks at the cross track error $e_e$. The cross track error $e_e$ is defined as the distance between the reference point and the closest point on our trajectory.
+
+![Stanley error with heading and cross track error](../assets/stanleyerror.png)
+*source: [2]*
+
+The first part of our steering angle tries to correct for this error $arctan(\frac{k_e * e_e}{k_v*v})$ while the second part just corrects for our heading error $\psi$.
+
+This leads to the following equation for our steering angle.
+
+$$
+\delta(t) = \psi(t) + arctan(\frac{k_e*e_e(t)}{k_v*v(t)})
+$$
+
+With $k_e$ and $k_v$ being tuenable parameters for cross tracking error and speed respectively.
```

Comment:
tuneable

Reply:
Fixed in fd9e0bc59df8f9825da05b888d6c2d2cd19fb2e4